packages/firebase_ml_vision/example/lib/main.dart

```diff
@@ -2,32 +2,151 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
+import 'dart:io';
+
 import 'package:flutter/material.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:firebase_ml_vision/firebase_ml_vision.dart';
 
-void main() => runApp(new MyApp());
+void main() => runApp(new MaterialApp(home: _MyHomePage()));
 
-class MyApp extends StatefulWidget {
+class _MyHomePage extends StatefulWidget {
   @override
-  _MyAppState createState() => new _MyAppState();
+  _MyHomePageState createState() => new _MyHomePageState();
 }
 
-class _MyAppState extends State<MyApp> {
-  @override
-  void initState() {
-    super.initState();
+class _MyHomePageState extends State<_MyHomePage> {
+  File _imageFile;
+  Size _imageSize;
+  List<TextBlock> _textLocations;
+
+  Future<void> _getImage() async {
+    setState(() {
+      _imageFile = null;
+      _imageSize = null;
+      _textLocations = null;
+    });
+
+    final File imageFile =
+        await ImagePicker.pickImage(source: ImageSource.camera);
+
+    setState(() {
+      _imageFile = imageFile;
+    });
+  }
+
+  Future<void> _scanImage() async {
+    final FirebaseVisionImage visionImage =
+        FirebaseVisionImage.fromFile(_imageFile);
+    final TextDetector detector = FirebaseVision.instance.getTextDetector();
+    final List<TextBlock> blocks = await detector.detectInImage(visionImage);
+
+    final Image image = Image.file(_imageFile);
+    final Size imageSize = await _getImageSize(image);
+
+    setState(() {
+      _textLocations = blocks;
+      _imageSize = imageSize;
+    });
+  }
+
+  Future<Size> _getImageSize(Image image) {
```

Comment:
Nit: Maybe this should be _getImageDimenisons

Reply:
Normally I prefer the word dimensions over size, but the method here actually returns a Size object. I think it makes more sense to use size in this context with this being the case.